src/config/source/command_line_arguments.cpp

```diff
@@ -0,0 +1,158 @@
+#include "config/source/command_line_arguments.h"
+
+#include <map>
+
+#include <fmt/format.h>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include "config/config_exception.h"
+#include "silo/common/panic.h"
+
+namespace silo::config {
+
+std::string CommandLineArguments::configKeyPathToString(const ConfigKeyPath& key_path) {
+   std::vector<std::string> result;
+   for (const auto& sublevel : key_path.getPath()) {
+      for (const std::string& current_string : sublevel) {
+         result.push_back(current_string);
+      }
+   }
+   return "--" + boost::join(result, "-");
+}
+
+AmbiguousConfigKeyPath CommandLineArguments::stringToConfigKeyPath(const std::string& option) {
+   if (option.size() < 3 || !option.starts_with("--")) {
+      throw silo::config::ConfigException(fmt::format(
+         "the provided option '{}' is not a valid command line option"
+         " as silo currently only accepts long-form options starting with '--'",
+         option
+      ));
+   }
+   std::string trimmed = option.substr(2);
+
+   std::vector<std::string> delimited_strings;
+   std::vector<std::string> tokens;
+
+   boost::split(tokens, trimmed, boost::is_any_of("-"));
+
+   for (const auto& token : tokens) {
+      if (token.empty()) {
+         throw silo::config::ConfigException(fmt::format(
+            "the provided option '{}' is not a valid command line option"
+            " because it contains an empty string segment between '-'",
+            option
+         ));
+      }
+      delimited_strings.push_back(token);
+   }
+
+   auto result = AmbiguousConfigKeyPath::tryFrom(std::move(delimited_strings));
+   if (result == std::nullopt) {
+      throw silo::config::ConfigException(
+         fmt::format("the provided option '{}' is not a valid command line option", option)
+      );
+   }
+   return result.value();
+}
+
+namespace {
+
+// Split an individual command line argument in option and value
+// parts, if a '=' was given.
```

Comment:
I've replaced it with a more useful comment giving examples. I felt it warranted explanation since with all the different ways in how configuration options are parsed, and given that I first forgot to even implement the "=" case, someone reading the code would be lost about what this does.